Refactor ToolsPage class to use esc_html_e and esc_attr_e for better localization support

hiroshisatoy opened the pull request:
Refactor ToolsPage class to use esc_html_e and esc_attr_e for better localization support

### Description of the Change

This PR fixes parts of the `ToolsPage` class where translation functions were not being used, ensuring proper translation support. Specifically, translation functions were added to hardcoded strings such as button value attributes, and the code was refactored to use `esc_html_e()` and `esc_attr_e()` for improved code consistency.

**Changes made:**
- Replaced `echo esc_html( __( 'Block Catalog - Index', 'block-catalog' ) )` with `esc_html_e( 'Block Catalog - Index', 'block-catalog' )` (code simplification)
- Replaced `echo esc_html( __( 'Select Post Type(s)', 'block-catalog' ) )` with `esc_html_e( 'Select Post Type(s)', 'block-catalog' )` (code simplification)
- Replaced `echo esc_html( __( 'Errors', 'block-catalog' ) )` with `esc_html_e( 'Errors', 'block-catalog' )` (code simplification)
- Added translation functions to button `value` attributes (previously hardcoded strings):
  - "Index Posts" button → `esc_attr_e( 'Index Posts', 'block-catalog' )`
  - "Delete Index" button → `esc_attr_e( 'Delete Index', 'block-catalog' )`
  - "Cancel" buttons (2 instances) → `esc_attr_e( 'Cancel', 'block-catalog' )`
- Fixed minor indentation inconsistencies

**Benefits:**
- Button labels and other text can now be properly translated
- Improved code readability and maintainability
- Better alignment with WordPress coding standards
- More consistent use of WordPress escaping functions

Closes #

### How to test the Change

1. Navigate to **Tools > Block Catalog** in the WordPress admin
2. Verify that all text displays correctly:
   - Page heading "Block Catalog - Index"
   - "Select Post Type(s)" heading
   - Button labels: "Index Posts", "Delete Index", and "Cancel"
   - Error messages (if any occur during indexing)
3. Test the indexing functionality to ensure buttons work correctly
4. If using a non-English language, verify that button labels are properly translated
5. Check browser console for any JavaScript errors

### Changelog Entry

Developer - Added translation functions to button labels and refactored ToolsPage class to use esc_html_e() and esc_attr_e() for improved localization support and code consistency.

### Credits

Props @sato-jp

### Checklist:
- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added [Critical Flows, Test Cases, and/or End-to-End Tests](https://10up.github.io/Open-Source-Best-Practices/testing/) to cover my change.
- [x] All new and existing tests pass.

hiroshisatoy committed:
Refactor ToolsPage class to use esc_html_e and esc_attr_e for better localization support

peterwilsoncc commented on includes/classes/ToolsPage.php:
```diff
@@ -90,31 +90,31 @@ public function render() {
 		<div id="index-status" style="display:none">
 			<progress class="index-progress-bar" id="index-progress" value="50" max="100">
-	    </progress>
+		</progress>
```
This is intended with spaces which is causing the coding standards sniff to fail.

I think you can just move the closing element to be on the same line as the opening element.

:1234: This applies for the same change below so I won't repeat myself.

hiroshisatoy replied:
Thank you for the review!

I've fixed the progress tag indentation issue by moving the closing tags to the same line as the opening tags, as you suggested. The changes have been committed.

I apologize if I made any mistakes with the GitHub workflow, as I'm still getting familiar with it. Please let me know if there's anything else that needs to be addressed.

peterwilsoncc reviewed (commented):
I've added a note inline re: the changes to the `<progress>` elements.

The i18n changes look good to me, thanks.

hiroshisatoy committed:
Fix progress tag indentation to resolve coding standards issue

Move closing progress tags to same line as opening tags to fix
indentation issues that were causing coding standards sniff failures.

hiroshisatoy committed:
Merge branch 'develop' into develop

peterwilsoncc reviewed (approved):
@sato-jp There's no need to apologize, you weren't to know and it's an easy fix.

These changes look good to me and test well. I've confirmed the JavaScript is not string dependent so that additional translations won't cause any issues.
